src/bson.ts

```diff
@@ -69,3 +69,20 @@ export function pluckBSONSerializeOptions(options: BSONSerializeOptions): BSONSe
     raw
   };
 }
+
+// Merge the given BSONSerializeOptions, preferring options over parentOptions, and substituting
+// defaults for values not set.
+export function inheritOrDefaultBSONSerializableOptions(
```

Comment:
The comments above are useful, nice work you can go ahead and make it a docstyle comment 
```
/** 
 * ...
 * @internal
 */
```
Then it will be readable by just hovering it anywhere in the codebase, also you can make it internal.

Also fine with it as is but may I suggest the name: `inheritBSONOptions` I think its fine to just state they will be defaulted in the docs.